Fix #7703:  ADD PAGE in Rocket Insights tab redirects to a blank page when URL limit is reached

Miraeld opened the pull request:
# Description

Fixes #7703 
Avoid the blank page when button should be disabled.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue).

## Detailed scenario

### What was tested

Refer to the issue.

### How to test

Refer to the issue.

## Technical description

### Documentation
This pull request makes a minor adjustment to the `handleAddPage` function in `src/js/global/ajax.js` to ensure that the default event behavior is prevented before any other logic is executed. This helps avoid unintended form submissions or navigation when the function is triggered.
### New dependencies

None
### Risks

None
# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style

- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

Miraeld committed:
Fix: Prevent default action in handleAddPage function for better UX

Miraeld committed:
Update js files

hanna-meda commented:
@Miraeld, thank you for this PR.

This issue can still be reproduced by steps:
1. As free user, add a 3rd url
2. Click on the "Add page" (either while the URL is still processing or when done, not relevant)
3. Notice a :no_entry_sign: sign but the redirect to a blank page does occur
4. Manual refresh the page, notice that now there is Max URL limit reached tooltip, no longer a redirection


https://github.com/user-attachments/assets/c07fda5d-17c8-4298-9e47-b26f2d0dc408

Miraeld commented:
https://github.com/user-attachments/assets/b73543f6-2e3b-4bdd-a460-2f61a25f0c94

I can't reproduce it...  However I tried something in the latest commit, could you check ?

Miraeld committed:
Enhance error handling for URL limit in Performance Monitoring

- Added 'remaining_urls' and 'can_add_pages' fields to the JSON error response when the maximum number of URLs is reached.
- Updated the AJAX error handling in the JavaScript to clear the input field, disable URL addition elements, and show a quota banner when the URL limit is reached.

Miraeld committed:
fix phpcs

wordpressfan committed:
Merge branch 'feature/3.20' into enhancement/7703-avoid-refresh-page

# Conflicts:
#	assets/js/wpr-admin.js
#	assets/js/wpr-admin.min.js
#	assets/js/wpr-admin.min.js.map

wordpressfan committed:
fix conflicts

hanna-meda commented:
@Miraeld, this can no longer be reproduced with the latest commit. Everything is working as expected now. Merging and closing. Thanks.
